src/Sulu/Bundle/AdminBundle/Resources/public/js/aura_extensions/sulu-buttons.js

```diff
@@ -0,0 +1,197 @@
+                }
+            };
+
+            app.sandbox.sulu.buttons.saveWithOptions = app.sandbox.util.extend(true, {}, app.sandbox.sulu.buttons.save, {
+                dropdownItems: ['saveBack', 'saveNew'],
+                dropdownOptions: {
+                    onlyOnClickOnArrow: true
+                }
+            });
+            /**
+             * Buttons definition (end)
+             */
+
+            /**
+             * Dropdown-items definition (start)
+             */
+            app.sandbox.sulu.buttons.dropdownItems = {};
+
+            app.sandbox.sulu.buttons.dropdownItems.smallThumbnails = {
+                id: 'small-thumbnails',
+                title: 'sulu.toolbar.small-thumbnails',
+                callback: function () {
+                    app.sandbox.emit('sulu.toolbar.change.thumbnail-small');
+                }
+            },
+
+            app.sandbox.sulu.buttons.dropdownItems.bigThumbnails = {
+                id: 'big-thumbnails',
+                title: 'sulu.toolbar.big-thumbnails',
+                callback: function () {
+                    app.sandbox.emit('sulu.toolbar.change.thumbnail-large');
+                }
+            },
+
+            app.sandbox.sulu.buttons.dropdownItems.table = {
+                id: 'table',
+                title: 'sulu.toolbar.table',
+                callback: function () {
+                    app.sandbox.emit('sulu.toolbar.change.table');
+                }
+            },
+
+            app.sandbox.sulu.buttons.dropdownItems.saveBack = {
+                id: 'save-back',
+                title: 'public.save-and-back',
+                callback: function () {
+                    app.sandbox.emit('sulu.toolbar.save', 'back');
+                }
+            },
+
+            app.sandbox.sulu.buttons.dropdownItems.saveNew = {
+                id: 'save-new',
+                title: 'public.save-and-new',
+                callback: function () {
+                    app.sandbox.emit('sulu.toolbar.save', 'new');
+                }
+            },
+
+            app.sandbox.sulu.buttons.dropdownItems.delete = {
+                id: 'delete-button',
+                title: 'public.delete',
+                callback: function () {
+                    app.sandbox.emit('sulu.toolbar.delete');
+                }
+            },
+            /**
+             * Dropdown-items definition (end)
+             */
+
+
+            /**
+             * Takes arguments and returns an array of toolbar-buttons
+             * An argument can be a string (a defined button) or an object which
+             * overrides different properties of a button
+             * @example
+             *
+             *      sulu.buttons.get('save', 'edit', {'save': {callback: myNewCallbackFunction}});
+             *
+             * @returns {Array} an array of buttons
+             */
+            app.sandbox.sulu.buttons.get = function() {
```

Comment:
i would probably create a function for `add(name, button)` and `get(name)` and save all buttons in a private array here ...